core/internal/runconsolelogs/runconsolelogs.go

```diff
@@ -79,20 +82,26 @@ func New(params Params) *Sender {
 		fsWriter = &filestreamWriter{FileStream: params.FileStreamOrNil}
 	}
 
-	fileWriter, err := NewOutputFileWriter(
-		filepath.Join(
-			params.Settings.GetFilesDir(),
-			string(params.ConsoleOutputFile),
-		),
-		params.Logger,
-	)
+	var fileWriter *outputFileWriter
+	if params.EnableCapture {
+		var err error
+		fileWriter, err = NewOutputFileWriter(
+			filepath.Join(
+				params.FilesDir,
+				string(params.ConsoleOutputFile),
+			),
+			params.Logger,
+		)
 
-	if err != nil {
-		params.Logger.CaptureError(
-			fmt.Errorf(
-				"runconsolelogs: cannot write to file: %v",
-				err,
-			))
+		if err != nil {
+			params.Logger.CaptureError(
+				fmt.Errorf(
+					"runconsolelogs: cannot write to file: %v",
+					err,
+				))
+		}
+	} else {
+		fileWriter = nil
```

Comment:
this isn't necessary, it will be nil if not initialized

Reply:
good point!